lib/src/code_generator/global.dart

```diff
@@ -42,12 +59,9 @@ class Global extends LookUpBinding {
       s.write(makeDartDoc(dartDoc!));
     }
 
-    final holderVarName =
-        w.wrapperLevelUniqueNamer.makeUnique('_$globalVarName');
-    s.write(
-        '${w.ffiLibraryPrefix}.Pointer<${type.getCType(w)}> $holderVarName;\n');
+    final refOrValue = type.broadType == BroadType.Struct ? 'ref' : 'value';
     s.write(
-        "${type.getDartType(w)} get $globalVarName => ($holderVarName ??= ${w.dylibIdentifier}.lookup<${type.getCType(w)}>('$originalName')).value;\n\n");
+        "late final ${type.getDartType(w)} $globalVarName = ${w.dylibIdentifier}.lookup<${type.getCType(w)}>('$originalName').$refOrValue;\n\n");
```

Comment:
I think global variables are not constants, they can be changed.

If that's the case, we should instead add getters and setters for this.

So for an int global variable, we can do this -
```dart
  late final ffi.Pointer<ffi.Int32> _globalVar =
      _dylib.lookup<ffi.Int32>('globalVar');

  int get globalVar => _globalVar.value;

  set globalVar(int value) => _globalVar.value = value;

```

Reply:
Yes, this works for all types except structs, which just don't have a setter, but fields for structs can be set from the ref that is obtained via the getter.